packages/react-dom/src/events/DOMModernPluginEventSystem.js

```diff
@@ -308,12 +326,62 @@ export function dispatchEventForPluginEventSystem(
   );
 }
 
+function getNearestRootOrPortalContainer(instance: Element): Element {
+  let node = getClosestInstanceFromNode(instance);
+  while (node !== null) {
+    const tag = node.tag;
+    // Once we encounter a host container or root container
+    // we can return their DOM instance.
+    if (tag === HostRoot || tag === HostPortal) {
+      return node.stateNode.containerInfo;
+    }
+    node = node.return;
+  }
+  return instance;
```

Comment:
Is this sound? If we don't find a host or portal root, can we just return the closest instance safely? (I don't have a good mental model of this. Just asking.)

Would we ever expect to encounter this other than the `window` or `document`?

**Edit** I guess, based on your comments below, this fallback is ok.

Reply:
We'll never encounter this with `window` or `document`. I've revised the logic to help make that clear.